src/services/bundle-descriptor-converter-service.ts

```diff
@@ -85,11 +87,31 @@ export class BundleDescriptorConverterService {
   }
 
   private generateMicroFrontendYamlDescriptor(microFrontend: MicroFrontend) {
+    const customUiFile = path.resolve(
+      MICROFRONTENDS_FOLDER,
+      microFrontend.name,
+      `${microFrontend.name}${CUSTOM_UI_EXTENSION}`
+    )
+    const customUiFileExists = fs.existsSync(customUiFile)
+
     const widgetDescriptor:
       | YamlWidgetDescriptor
       | YamlWidgetConfigDescriptor
-      | YamlAppBuilderWidgetDescriptor =
-      this.mapMicroFrontendToYamlDescriptor(microFrontend)
+      | YamlAppBuilderWidgetDescriptor = this.mapMicroFrontendToYamlDescriptor(
+      microFrontend,
+      customUiFileExists
+    )
+
+    if (customUiFileExists) {
```

Comment:
In case FTL file is found, I would add an info log printing: "Custom widget template FTL found, including it"